sql/core/src/main/scala/org/apache/spark/sql/execution/ExternalAppendOnlyUnsafeRowArray.scala

```diff
@@ -138,6 +142,7 @@ private[sql] class ExternalAppendOnlyUnsafeRowArray(
           initialSize,
           pageSizeBytes,
           numRowsSpillThreshold,
+          maxSizeSpillThreshold,
```

Comment:
looking at this `def add` method, it trigger spilling only if the num rows exceeds `numRowsInMemoryBufferThreshold`. IIUC we are not using memory-based threshold here, as we keep appending data to a memory buffer based on the num rows threshold.